data/common.yaml

```diff
@@ -1 +1,3 @@
 ---
+k8s::version: 1.26.1
+k8s::etcd_version: 3.5.1
```

Comment:
The values were moved from init.pp to support being found with a `lookup()` call.  
`k8s::version` could probably stay were is is, but `k8s::etcd_version` would either need to be hieradata, be duplicated, or be entirely hidden away under `k8s::server::etcd::version`.

I am not sure which of these cases can be considered the best in terms of discoverability and manageability.

Reply:
Maybe a few things to consider that can help to avoid this situation where user-supplied values are ignored:

1. Make the 'root' class (`k8s`) something that shall always be in the user's catalog, then use `$k8s::version` and `$k8s::etcd_version` where needed;
2. Variant of the above, have a `k8s::globals` class which is always required by the other classes and allow to set the default values (example: puppetlabs-postgresql);
3. Split `etcd` as another module, and where is is used in the `k8s` module `include etcd` and then use `$etcd::version`.

Reply:
Splitting all actual functionality out of the main `k8s` class has been a consideration for a while, but also something that wasn't really worked on as it was a minor issue - since strict variables wasn't really a problem and therefore a few `pick()` calls solved that need.  
Having etcd as a separate module was actually how this module was designed originally - before it became `k8s`, but it eventually got merged into one due to the respectively tiny amount of unique code in the etcd module, especially compared with the large amount of duplicated code between the two.

Separating out all the user/folder/repo/etc handling from the main `k8s` class and ensuring that it's included when necessary will probably be the best way to go about things, going to push another commit tomorrow towards that direction.